File: web/src/api/group.js
import service from '@/utils/request'

// 获取分组树
export const getGroupTree = () => {
  return service({
    url: '/group/tree',
    method: 'get'
  })
}

// 获取分组列表
export const getGroupList = (params) => {
  return service({
    url: '/group/list',
    method: 'get',
    params
  })
}

// 创建分组
export const createGroup = (data) => {
  return service({
    url: '/group',
    method: 'post',
    data
  })
}

// 更新分组
export const updateGroup = (id, data) => {
  return service({
    url: `/group/${id}`,
    method: 'put',
    data
  })
}

// 删除分组
export const deleteGroup = (id) => {
  return service({
    url: `/group/${id}`,
    method: 'delete'
  })
}

File: server/router/gb28181/group.go
package gb28181

import (
	"wvp-go/server/api/v1/gb28181"

	"github.com/gin-gonic/gin"
)

type GroupRouter struct{}

var groupRouter = new(GroupRouter)

func (g *GroupRouter) InitGroupRouter(Router *gin.RouterGroup) {
	r := Router.Group("/group")
	{
		r.GET("/tree", gb28181.GetGroupTree)
		r.GET("/list", gb28181.GetGroupList)
		r.POST("", gb28181.CreateGroup)
		r.PUT("/:id", gb28181.UpdateGroup)
		r.DELETE("/:id", gb28181.DeleteGroup)
	}
}
